python/pyiceberg/avro/writer.py

```diff
@@ -193,6 +195,8 @@ class MapWriter(Writer):
 
     def write(self, encoder: BinaryEncoder, val: Dict[Any, Any]) -> None:
         encoder.write_int(len(val))
-        for k, v in val:
+        for k, v in val.items():
             self.key_writer.write(encoder, k)
             self.value_writer.write(encoder, v)
+        if len(val) > 0:
+            encoder.write_int(0)
```

Comment:
From https://avro.apache.org/docs/1.11.1/specification/#schema-maps : "A block with count zero indicates the end of the map."